inc/common/purge.php

```diff
@@ -420,16 +420,18 @@ function rocket_post_purge_urls_for_qtranslate( $urls ) {
  */
 function do_admin_post_rocket_purge_cache() { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 	if ( isset( $_GET['type'], $_GET['_wpnonce'] ) ) {
-		$type = sanitize_key( $_GET['type'] );
+		$type       = sanitize_key( $_GET['type'] );
+		$type_array = explode( '-', $type );
 
-		$_type     = explode( '-', $type );
-		$_type     = reset( $_type );
-		$_id       = explode( '-', $type );
-		$_id       = end( $_id );
-		$_taxonomy = isset( $_GET['taxonomy'] ) ? sanitize_title( wp_unslash( $_GET['taxonomy'] ) ) : false;
+		$_type     = reset( $type_array );
+		$_id       = end( $type_array );
+		$_id       = is_numeric( $_id ) ? absint( $_id ) : 0;
+		$_taxonomy = isset( $_GET['taxonomy'] ) ? sanitize_title( wp_unslash( $_GET['taxonomy'] ) ) : '';
+		$_url      = '';
 
 		if ( ! wp_verify_nonce( sanitize_key( $_GET['_wpnonce'] ), 'purge_cache_' . $type ) ) {
 			wp_nonce_ays( '' );
+			return;
```

Comment:
Does it need a return after wp_nonce_ays()?

Reply:
@crystinutzaa Yes, for the same reason we need it after `wp_die()`: to make it testable. Well, even if I admit I didn't created a test for this case ^^'